node/core/approval-voting/src/import.rs

```diff
@@ -82,14 +83,36 @@ struct ImportedBlockInfoEnv<'a> {
 	keystore: &'a LocalKeystore,
 }
 
-// Computes information about the imported block. Returns `None` if the info couldn't be extracted -
-// failure to communicate with overseer,
+#[derive(Debug, thiserror::Error)]
+enum ImportedBlockInfoError {
+	// NOTE: The `RuntimeApiError` already prints out which request it was,
+	//       so it's not necessary to include that here.
+	#[error(transparent)]
+	RuntimeError(RuntimeApiError),
+
+	#[error("future cancalled while requesting {0}")]
+	FutureCancelled(&'static str, futures::channel::oneshot::Canceled),
```

Comment:
`Canceled` should be annotated with `#[source]`

Reply:
Ah, I missed that.

That said, in this case in practice since `Canceled` is a ZST and doesn't actually carry any information it shouldn't make any practical difference.

I'm happy to make a followup PR and add this in though.